Introduce g:gen_gtags#gtags_bin and g:gen_gtags#gtags_opts

jetm opened the pull request:
g:gen_gtags#gtags_bin allows to use a different gtags binary file.

g:gen_gtags#gtags_opts allows to insert gtags options before the path to
database.

jsfaint commented:
Thanks for the PR.
There is another command `global` used in gtags support which is also belong to GNU global too.

Please add support for it too:smile:

jetm commented:
@jsfaint Updated the PR.

jsfaint commented:
![image](https://user-images.githubusercontent.com/571829/40264556-1766fa7a-5b59-11e8-9f16-c0f269a48b7b.png)
I tried the patch, it looks the `g:gen_gtags#gtags_opts` cause some issue when generate the gtags

After some modify 
```vim
let l:cmd = [g:gen_tags#gtags_bin, g:gen_tags#gtags_opts, l:db_dir]
```
to 
```vim
let l:cmd = [g:gen_tags#gtags_bin, l:db_dir, g:gen_tags#gtags_opts]
```
It works well.
It looks gtags will treat `''` empty parameter as directory.

Another question, what value of `g:gen_tags#gtags_opts` will you use?

Thanks

Commit:
Introduce gtags_bin, global_bin and gtags_opts

g:gen_gtags#gtags_bin allows to use a different gtags binary file.

g:gen_gtags#global_bin allows to use a different global binary file.

g:gen_gtags#gtags_opts allows to insert gtags options before the path to
database.

jetm commented:
Thank you for the catch. Submitted your suggestion. 
A the beginning I didn't test it because the gtags help says `gtags [-f file][dbpath]`. I used the same order.

The initial problem I faced was how to tell to gtags that filter by files. I develop on a large codebase. A filter reduces the tag file from 6 GB to 1.6 GB and also take less time. The option I pass is this one:
```vim
let s:tags_files = expand('$HOME/tmp/tags.files')
let g:gen_tags#ctags_opts = "-L" . s:tags_files
let g:gen_tags#gtags_opts = "-f" . s:tags_files
```
I use the equivalent for ctags. I generate manually, using a script, the _tags.files_ file. Another feature could be to generate that file from Vim.

jsfaint commented:
Got it! Thanks for the explanation.
Merged!

If you finished the filter feature, can you make another PR?
There are another issue #44 , request this feature too.
Thank you! 👍

jetm commented:
Thanks for the merge!
Right now to filter by files in `ctags` and `gtags` I have a _hack_ that is not ready for a PR. Let me see between all the things I need to do, I can polish it and make a PR. Thanks!
